packages/amplication-build-manager/src/utils.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@nestjs/common";
+import { promises as fs } from "fs";
+
+@Injectable()
+export class UtilsService {
+  /**
+   * This function extracts the buildId from the buildId with suffix.
+   * It's needed because the buildId is used as a key in the Kafka messages and and as a folder name in the artifacts and when it
+   * interacts with the data-service-generator and the server it needs to be without the suffix.
+   * @param buildIdWithSuffix the buildId with suffix which in this case could be "-server" or "-admin-ui"
+   * @returns return the substring before the first hyphen or the whole string if there is no hyphen
+   */
+  extractBuildId(buildIdWithSuffix: string): string {
+    const hyphenIndex = buildIdWithSuffix.indexOf("-");
```

Comment:
Even if this seems ok now, it might hurt us in the future as in case of buildId format change (i.e. `asd123dab40s-adfnv0`).

I'd suggest to be more strict and check for the expected child buildId. 
i.e. if you have a rule inside an hypothetical  `getChildrenBuildId(buildId)` that uses `-server` and `-admin`, use that to retrieve the partentBuildId